Makefile

```diff
@@ -12,25 +12,70 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+TOOLS_DIR := hack/tools
+BIN_DIR := bin
+TOOLS_BIN_DIR := $(TOOLS_DIR)/$(BIN_DIR)
+
+GOTESTSUM := $(TOOLS_BIN_DIR)/gotestsum
+GOLANGCI_LINT := $(TOOLS_BIN_DIR)/golangci-lint
+
+export GO111MODULE=on
+
+.PHONY: all
 all: build
 
 # TODO(ixdy): containerize
+.PHONY: build
 build:
 	go build ./...
 
-test:
-	go test ./...
+.PHONY: test
+test: $(GOTESTSUM)
+	$(GOTESTSUM) $${ARTIFACTS:+--junitfile="${ARTIFACTS}/junit.xml"} ./...
```

Comment:
TILd

Reply:
this is maybe too clever, but it works for now...